src/internal/git/pull.go

```diff
@@ -1,87 +1,84 @@
 package git
 
 import (
-	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
 	"regexp"
 
 	"github.com/defenseunicorns/zarf/src/config"
 	"github.com/defenseunicorns/zarf/src/internal/message"
 	"github.com/defenseunicorns/zarf/src/internal/utils"
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing"
-
-	"strings"
 )
 
 const onlineRemoteName = "online-upstream"
 
-func DownloadRepoToTemp(gitUrl string, spinner *message.Spinner) string {
+// DownloadRepoToTemp clones or updates a repo into a temp folder to perform ephemeral actions (i.e. process chart repos).
+func DownloadRepoToTemp(gitURL string, spinner *message.Spinner) string {
 	path, _ := utils.MakeTempDir(config.CommonOptions.TempDirectory)
 	// If downloading to temp, grab all tags since the repo isn't being
 	// packaged anyway, and it saves us from having to fetch the tags
 	// later if we need them
-	pull(gitUrl, path, spinner)
+	pull(gitURL, path, spinner, "")
 	return path
 }
 
-func Pull(gitUrl, targetFolder string, spinner *message.Spinner) (string, error) {
-	repoName, err := transformURLtoRepoName(gitUrl)
+// Pull clones or updates a git repository into the target folder.
+func Pull(gitURL, targetFolder string, spinner *message.Spinner) (string, error) {
+	repoName, err := transformURLtoRepoName(gitURL)
 	if err != nil {
-		message.Errorf(err, "unable to pull the git repo at %s", gitUrl)
+		message.Errorf(err, "unable to pull the git repo at %s", gitURL)
 		return "", err
 	}
 
 	path := targetFolder + "/" + repoName
-	pull(gitUrl, path, spinner)
+	pull(gitURL, path, spinner, repoName)
 	return path, nil
 }
 
-func pull(gitUrl, targetFolder string, spinner *message.Spinner) {
-	spinner.Updatef("Processing git repo %s", gitUrl)
-
-	gitCred := FindAuthForHost(gitUrl)
-
-	matches := strings.Split(gitUrl, "@")
-	onlyFetchRef := len(matches) == 2
-	cloneOptions := &git.CloneOptions{
-		URL:        matches[0],
-		Progress:   spinner,
-		RemoteName: onlineRemoteName,
-	}
+func pull(gitURL, targetFolder string, spinner *message.Spinner, repoName string) {
+	spinner.Updatef("Processing git repo %s", gitURL)
 
-	if onlyFetchRef {
-		cloneOptions.Tags = git.NoTags
+	gitCachePath := targetFolder
+	if repoName != "" {
+		gitCachePath = filepath.Join(config.GetCachePath(), fmt.Sprintf("repos/%s", repoName))
```

Comment:
Like the `images` directory in `images/pull.go` could we make this a const? Having a hardcoded string for directory traversing seems sketchy.

Even if not, we should be using a nested `filepath.Join(..)` instead of doing `fmt.Sprintf("repos/%s")`

Reply:
done